### PR TITLE
Fix: T_ManagedExec.ExecuteBinaryDoubleFork Crashes Hard

### PR DESCRIPTION
**Pull request by reneme**
While testing some Jenkins specific things @Moliholi found that `T_ManagedExec.ExecuteBinaryDoubleFork` fails occasionally  with *SIGPIPE*. This test double forks and executes `gdb` and tries to communicate with the new process through a pipe. Turns out, that `gdb` was not installed on the machine.

I suppose, that there is a race between the parent process and the forked one when the parent is writing to the pipe. If the child process didn't invoke `execvp(gdb)` yet, it will still be around and the write to the pipe works fine. However, if the child already invoked `execvp(gdb)` (and thus is dead already since `gdb` is not found) the parent receives *SIGPIPE* because of the write to a broken pipe and crashes hard as well.

I added `GetExecutablePath()` to `util.h` that resembles the `which` utility in bash. Using that, the test case can check if `gdb` is installed before trying to execute it. Furthermore there is a test `T_Util.GetExecutablePath` to verify the function.

### CODE REVIEW

**Comment by jblomer** [comment]
Since `GetExecutablePath()` is currently only needed by the tests, it should rather be placed in test_util for the time being.

File: cvmfs/util.cc [hunk 1]
```diff
@@ -1373,6 +1373,64 @@ bool ManagedExec(const vector<string>  &command_line,
...
+    return result;
+  }
+
+  const std::vector<std::string> paths = SplitString(searchpath, ':');
+        std::vector<std::string>::const_iterator i    = paths.begin();
+  const std::vector<std::string>::const_iterator iend = paths.end();
+  for (; i != iend; ++i) {
+    const std::string candidate_path = *i + "/" + exe_name;
+
+    char *real_path = realpath(candidate_path.c_str(), NULL);
```
Comment (jblomer) [inline cvmfs/util.cc]: Real path can be put on the stack as char[MAX_PATH], which should avoid a bunch of `malloc()` and `free()` calls.

File: cvmfs/util.cc [hunk 2]
```diff
@@ -1373,6 +1373,64 @@ bool ManagedExec(const vector<string>  &command_line,
...
+    if (NULL == real_path) {
+      continue;
+    }
+
+    struct stat statinfo;
+    const int res = stat(real_path, &statinfo);
```
Comment (jblomer) [inline cvmfs/util.cc]: `stat()` in cvmfs is handled by the portable (64bit libc interface on Linux and OS X) `platform_stat`

**Comment by reneme** [comment]
Agreed and [moved to testutil.h](https://github.com/reneme/cvmfs/commit/0bda5fad30348aa7d0b13e8ecaa3528c5c5200d8). Avoiding `malloc()` is a good idea [as well](https://github.com/reneme/cvmfs/commit/e179166eca7b94ff2b5435571f4ae4cc65922231). Meh, forgot about the platform `stat()`, [fixed](https://github.com/reneme/cvmfs/commit/8ab48c4cdb3af5313aea3f9f9cbc694eaf7cfcb8) as well.

**Pull request merged**